drivers/flash/nrf_qspi_nor.c

```diff
@@ -625,6 +647,10 @@ static int qspi_erase(const struct device *dev, uint32_t addr, uint32_t size)
 		return -EINVAL;
 	}
 
+	if (dev_data->xip_enabled) {
+		lock_key = irq_lock();
```

Comment:
This presents a problem then because when busy with an erase, it needs to block _all_ interrupts because one of them might be located in or call a function that resides in QSPI flash at which point the device will lock up